allensdk/brain_observatory/ecephys/current_source_density/_filter_utils.py

```diff
@@ -0,0 +1,79 @@
+from typing import List, Tuple
+import numpy as np
+
+from scipy import signal
+
+
+def select_good_channels(lfp: np.ndarray,
+                         reference_channels: List[int],
+                         noisy_channel_threshold: float
+                         ) -> Tuple[np.ndarray, np.ndarray]:
+    """Remove reference channels and channels that are too noisy from lfp data.
+
+    Parameters
+    ----------
+    lfp : numpy.ndarray
+        LFP data in the form of: trials x channels x time samples
+    reference_channels : List[int]
+        Reference channel indices for this probe.
+    noisy_channel_threshold : float
+        Lowest mean standard deviation that constitutes a "clean" LFP channel
+
+    Returns
+    -------
+    Tuple[cleaned_lfp, good_indices]
+        cleaned_lfp: numpy.ndarray
+            LFP where reference and noisy channels have been removed.
+            Data still in form of: trials x channel x time samples
+        good_indices: numpy.ndarray
+            Array of channel indices that are neither reference nor noisy.
+    """
+    channel_variance = np.mean(np.std(lfp, 2), 0)
+    noisy_channels = np.where(channel_variance > noisy_channel_threshold)[0]
+
+    to_remove = np.concatenate((np.array(reference_channels), noisy_channels))
+    good_indices = np.delete(np.arange(0, lfp.shape[1]), to_remove)
+
+    # Remove noisy or reference channels (axis=1)
+    cleaned_lfp = np.delete(lfp, to_remove, axis=1)
+
+    return (cleaned_lfp, good_indices)
+
+
+def filter_lfp_channels(lfp: np.ndarray,
```

Comment:
Better to make filter and them pass it into a function. This allows examining a filter or using something other than butterworth.

Reply:
Hm... since this is pipeline code (I think?), I don't think the filters would be switched out frequently enough to warrant a dependency injection approach?

Reply:
my understanding of AllenSDK that it is envisioned to be used as a general analysis toolkit.